qiita_db/data.py

```diff
@@ -1021,6 +1098,61 @@ class ProcessedData(BaseData):
     _preprocessed_processed_table = "preprocessed_processed_data"
 
     @classmethod
+    def get_by_status(cls, status):
+        """Returns id for all ProcessedData with given status
+
+        Parameters
+        ----------
+        status : str
+            Status to search for
+
+        Returns
+        -------
+        list of int
+            All the processed data id that match the given status
+        """
+        conn_handler = SQLConnectionHandler()
+        sql = """SELECT processed_data_id FROM qiita.processed_data pd
+                JOIN qiita.processed_data_status pds
+                    USING (processed_data_status_id)
+                WHERE pds.processed_data_status=%s"""
+        result = conn_handler.execute_fetchall(sql, (status,))
+        if result:
+            pds = [x[0] for x in result]
```

Comment:
Minor, but should this be a set? I doubt we could be getting repeated identifiers from the query above, but maybe.